### PR TITLE
improve perf of CopyToCurrentCache in post processing

### PR DESCRIPTION
**Pull request by hellosnow**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs [hunk 1]
```diff
@@ -187,27 +189,31 @@ private void CopyToCurrentCache(List<ManifestItem> increItems)
...
+                                      select oi;
+                Parallel.ForEach(
+                    itemsToBeCopied,
+                    new ParallelOptions { MaxDegreeOfParallelism = Environment.ProcessorCount },
+                    item =>
                     {
-                        throw new BuildCacheException($"Last incremental post processor outputs should contain {item.RelativePath}.");
-                    }
+                        string cachedFileName;
+                        if (!_increContext.LastInfo.PostProcessOutputs.TryGetValue(item.RelativePath, out cachedFileName))
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs]: is it thread safe?

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Threading.Tasks;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs]: move into namespace

File: src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs (hunk 1 above)
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/PostProcessors/PostProcessorsHandlerWithIncremental.cs]: at same time, will only get from dictionary without add or update, I think it's safe at this point.

---
In reply to: [107859187](https://github.com/dotnet/docfx/pull/1481#discussion_r107859187) [](ancestors = 107859187)

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**